paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -442,9 +440,9 @@ private Factory(Options options, RowType rowType, List<String> primaryKeys) {
                                     });
 
                     // add self
-                    sequenceFields.forEach(
-                            fieldName -> {
-                                int index = fieldNames.indexOf(fieldName);
+                    allSequenceFields.forEach(
```

Comment:
Here should use `sequenceFields` (the same reason that here is in a for loop).

Reply:
done.